### PR TITLE
[tvmc] add instruments for PassContext

### PR DESCRIPTION
**Pull request by zhyncs**
cc @leandron

### CODE REVIEW

File: python/tvm/driver/tvmc/compiler.py [hunk 1]
```diff
@@ -223,6 +223,7 @@ def compile_model(
     use_vm: bool = False,
     mod_name: Optional[str] = "default",
     workspace_pools: Optional[WorkspaceMemoryPools] = None,
+    instruments = None,
```
Comment (lhutton1) [inline python/tvm/driver/tvmc/compiler.py:226]: nit: could we add the `Optional[Sequence[PassInstrument]]` type annotation

**Review by lhutton1 — COMMENTED** [review]
Thanks for exposing this @zhyncs, overall LGTM, just had one small nitpick and looks like linting issues need addressing

File: python/tvm/driver/tvmc/compiler.py (hunk 1 above)
Comment (zhyncs) [reply python/tvm/driver/tvmc/compiler.py]: > nit: could we add the `Optional[Sequence[PassInstrument]]` type annotation

done

File: tests/python/driver/tvmc/test_compiler.py [hunk 2]
```diff
@@ -697,5 +697,36 @@ def test_compile_check_workspace_pools(mock_pkg, mock_fe, mock_relay):
...
+        def run_before_pass(self, mod, info):
+            self.run_before_count = self.run_before_count + 1
+
+        def run_after_pass(self, mod, info):
+            self.run_after_count = self.run_after_count + 1
+
+    passes_counter = PassesCounter()
+    tvmc_model = tvmc.load(keras_resnet50)
+    tvmc.compile(tvmc_model, target="llvm", instruments=[passes_counter])
+    assert passes_counter.run_after_count == passes_counter.run_before_count
```
Comment (Mousius) [inline tests/python/driver/tvmc/test_compiler.py:728]: Given `run_before_count` and `run_after_count` both start as `0`, wouldn't that mean that if we never used these instruments this assertion would still pass? Potentially worth asserting the actual number as well or that it's at `> 0` ?

Comment (zhyncs) [reply tests/python/driver/tvmc/test_compiler.py:719]: > Given `run_before_count` and `run_after_count` both start as `0`, wouldn't that mean that if we never used these instruments this assertion would still pass? Potentially worth asserting the actual number as well or that it's at `> 0` ?

done

**Comment by zhyncs** [comment]
The i386 compilation failure was caused by the git fetch error. How can I trigger a new one tvm-ci/pr-head? @lhutton1 @Mousius @gromero

**Comment by lhutton1** [comment]
@tvm-bot rerun

**Comment by zhyncs** [comment]
> @tvm-bot rerun

It seems that it doesn't work.

**Comment by lhutton1** [comment]
Hmmm seems like it didn't work for some reason, could you try a git commit --amend then a force push from your local branch?

**Comment by zhyncs** [comment]
https://ci.tlcpack.ai/blue/organizations/jenkins/tvm/detail/PR-13136/7/pipeline/520
It seems that the CI system is not stable and so slow.
image
May I trigger the failed `docs: GPU` separately without the whole process? @lhutton1 @Mousius @gromero

**Comment by zhyncs** [comment]
And the `unittest: CPU` failed with the `Upload JUnits to S3` error.
image

**Comment by driazati** [comment]
sorry the CI error there is a bit confusing (and you got a bit unlucky with the docs checkout failure, it should be fixed for subsequent runs), however there are test failures on arm: https://ci.tlcpack.ai/blue/organizations/jenkins/tvm/detail/PR-13136/7/pipeline/488#step-1582-log-1404

**Comment by areusch** [comment]
@zhyncs this looks like a github problem. they are reporting problems with their API servers, but not with git ops. might be worth retrying in a few hours in case there's something unreported going on.

**Comment by zhyncs** [comment]
> sorry the CI error there is a bit confusing (and you got a bit unlucky with the docs checkout failure, it should be fixed for subsequent runs), however there are test failures on arm: https://ci.tlcpack.ai/blue/organizations/jenkins/tvm/detail/PR-13136/7/pipeline/488#step-1582-log-1404

fixed

**Review by lhutton1 — APPROVED** [review]
Thanks @zhyncs, LGTM!

**Comment by zhyncs** [comment]
@leandron @lhutton1 @Mousius @gromero Could you merge this PR?

**Pull request merged**